src/tests/camera_settings.cpp

```diff
@@ -60,33 +75,61 @@ void CameraSettings::getAndSetPossibleOptions()
         Camera::Option set_option = getOption(_config.param_name);
         EXPECT_EQ(set_option, option);
     }
-    std::cout << "Resetting setting to initial option." << std::endl;
-    setOption(_config.param_name, initial_option);
 }
 
-void CameraSettings::setSettingWithSubSetting()
+void CameraSettings::setSomeRangeOptions()
 {
-    Camera::Option initial_option = getOption(_config.param_name);
-
-    std::cout << "Getting possible settings for " << _config.param_name << std::endl;
+    std::cout << "Getting range properties for " << _config.param_name << std::endl;
     std::vector<Camera::Option> options;
     _camera.get_possible_options(_config.param_name, options);
 
-    bool option_available = false;
-    for (const auto& option : options) {
-        if (option.option_id != _config.param_value) {
-            continue;
-        }
-        option_available = true;
-        setOption(_config.param_name, option);
+    ASSERT_EQ(options.size() >= 2, true);
+    ASSERT_EQ(options.size() <= 3, true);
 
-        // TODO: add support for Options with range type (e.g. wb-temp 4000..10000).
+    // FIXME: dealing with this interface is not great and needs improvement.
+
+    const auto& minimum = std::atoi(options[0].option_id.c_str());
```

Comment:
```suggestion
    int minimum = std::atoi(options[0].option_id.c_str());
```

Reply:
Ok the reference is a leftover from when it was a string and doesn't make sense actually. For `auto` vs. `int` I'm going to try more `auto` according to the C++ guidelines.

Reply:
Which guidelines? From my experience you shouldn't overdo it, because it quickly becomes hard to track the types and therefore harder to understand the code.

Reply:
You're right, it's not in the [CppCoreGuidelines](https://github.com/isocpp/CppCoreGuidelines/) (yet). It's just recommended by some of the guys (e.g. https://herbsutter.com/2013/08/12/gotw-94-solution-aaa-style-almost-always-auto/)

I suppose the idea is that your IDE should tell you the types when you want to know.

Reply:
He makes some valid arguments, but as he writes as well, there's a trade-off to readability. And looking at larger python projects, missing type information is one of the bigger issues I see there. So I'm not that convinced (yet).

Reply:
True, I tend to agree.